src/Logary/Scheduling.fs

```diff
@@ -4,74 +4,90 @@ module Logary.Internals.Scheduling
 
 // creds to Dave Thomas for his F# snippet
 open System.Threading
-  
-open FSharp.Actor
+
+open Hopac
+open Hopac.Infixes
 
 open NodaTime
 
+type Cancellation = private { cancelled: IVar<unit> }
+
+[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
+module Cancellation =
+  let create () =
+    { cancelled = IVar.Now.create () }
+
+  // https://github.com/logary/logary/pull/92#issuecomment-164139379
+  let isCancelled cancellation = job {
+    return! (cancellation.cancelled ^->. true) <|> (Alt.always false)
```

Comment:
This, using a promise (or ivar) as a cancellation token like thing, is actually one of the valid cases where one could use [isFulfilled](http://hopac.github.io/Hopac/Hopac.html#def:val Hopac.Promise.Now.isFulfilled).